### PR TITLE
remove genericRightPaneComponent and create RightPaneWrapper with form

### PR DESCRIPTION
**Pull request by hardiknai-techm**
Only below panel use `RightPaneForm`
CopyNotebookPane, 
PublishNotebookPaneAdapter, 
DeleteCollectionPanel, 
SettingPane, 
UploadFilePane,
UploadItemsPane,

after approve need to remove `genericRightPaneComponent ` and use 'RightPaneForm' in below panel 
ExecuteSprocParamsPanel, 
LoadQueryPanel, 
SaveQueryPanel, 
TableQuerySelectPanel

we can also use 'RightPaneForm' in below panel instend of three different import like `PanelInfoErrorComponent`, `PanelFooterComponent`, `PanelLoadingScreen` handle close logic separately.

AddCollectionPanel,
DeleteDatabaseConfirmationPanel

### CODE REVIEW

File: src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  isExecuting: boolean;
+  onClose: () => void;
+  onSubmit: () => void;
+  submitButtonText: string;
+  title: string;
+  isSubmitButtonHidden?: boolean;
+  children?: ReactNode;
+}
+
+export interface GenericRightPaneState {
```
Comment (victor-meng) [inline src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx:23]: RightPaneFormState

File: src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  const panelInfoErrorProps: PanelInfoErrorProps = {
+    messageType: "error",
+    message: formError,
+    formError: formError !== "",
+    showErrorDetails: formErrorDetail !== "",
+    openNotificationConsole: showErrorDetail,
+  };
+
+  return (
+    <div tabIndex={-1} onKeyDown={onKeyDown}>
```
Comment (victor-meng) [inline src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx:90]: I don't think we need any of the div layers above `panelContentWrapper` (line 90 - 92). The `PanelContainerComponent` already take cares of these.

These extra layers also mess up the styling of the panel.

**Review by victor-meng — CHANGES_REQUESTED** [review]

File: src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx (hunk 1 above)
Comment (hardiknai-techm) [reply src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx]: remove

File: src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx (hunk 2 above)
Comment (hardiknai-techm) [reply src/Explorer/Panes/RightPaneForm/RightPaneForm.tsx:86]: remove <div tabIndex={-1} onKeyDown={onKeyDown}></div>

**Review by southpolesteve — APPROVED** [review]

**Pull request merged**